dss/operations/iam.py

```diff
@@ -66,16 +67,51 @@ def get_fus_role_attached_policies(fus_client, action, role):
 def extract_aws_policies(action: str, client, managed: bool):
     """
     Call the AWS IAM API to retrieve policies and perform an action with them.
+
+    :param action: what action to take with the IAM policies (list, dump)
+    :param client: the boto client to use
+    :param managed: (boolean) if true, include AWS-managed policies
+    :returns: a list of items whose type depends on the action param
+        (policy names if action is list, json documents if action is dump)
     """
-    pass
+    master_list = []  # holds main results
+
+    if managed:
+        paginator_scope = "All"
+    else:
+        paginator_scope = "Local"
+
+    paginator = client.get_paginator("list_policies")
+    for page in paginator.paginate(Scope=paginator_scope):
+        for policy in page["Policies"]:
+
+            if action == "list":
+                policy_name = policy["PolicyName"]
+                master_list.append(policy_name)
+            elif action == "dump":
+                # first save as strings
+                master_list.append(json.dumps(policy, sort_keys=True, default=str))
+
+    if action == "list":
+        # Sort names, remove duplicates
+        master_list = sorted(list(set(master_list)))
+    elif action == "dump":
+        # Convert to strings, remove dupes, convert to back dicts
+        master_list = list(set(master_list))
+        master_list = [json.loads(j) for j in master_list]
```

Comment:
Can the set(master_list) be used in the list comprehension?

Reply:
👍  good call